drivers/bluetooth/hci/userchan.c

```diff
@@ -93,24 +93,70 @@ static bool is_hci_event_discardable(const uint8_t *evt_data)
 	}
 }
 
-static struct net_buf *get_rx(const uint8_t *buf)
+static struct net_buf *get_rx_evt(const uint8_t *data)
 {
-	bool discardable = false;
-	k_timeout_t timeout = K_FOREVER;
+	bool discardable = is_hci_event_discardable(data);
+	struct net_buf *buf;
+
+	do {
+		buf = bt_buf_get_evt(data[1], discardable, discardable ? K_NO_WAIT : K_SECONDS(10));
+		if (buf == NULL) {
+			if (discardable) {
+				LOG_DBG("Discardable buffer pool full, ignoring event");
+				return buf;
+			}
+			LOG_WRN("Couldn't allocate a buffer after waiting 10 seconds.");
+		}
+	} while (!buf);
 
-	switch (buf[0]) {
-	case BT_HCI_H4_EVT:
-		if (is_hci_event_discardable(buf)) {
-			discardable = true;
-			timeout = K_NO_WAIT;
+	return buf;
+}
+
+static struct net_buf *get_rx_acl(const uint8_t *data)
+{
+	struct net_buf *buf;
+
+	buf = bt_buf_get_rx(BT_BUF_ACL_IN, K_NO_WAIT);
+	if (buf == NULL) {
+		LOG_ERR("No available ACL buffers!");
+	}
+
+	return buf;
+}
+
+static struct net_buf *get_rx_iso(const uint8_t *data)
+{
+	static unsigned int fail_cnt;
+	struct net_buf *buf;
+
+	buf = bt_buf_get_rx(BT_BUF_ISO_IN, K_NO_WAIT);
+	if (buf == NULL) {
+		if (fail_cnt == 0) {
+			LOG_ERR("No available ISO buffers!");
 		}
 
-		return bt_buf_get_evt(buf[1], discardable, timeout);
+		fail_cnt++;
+	} else {
+		if (fail_cnt) {
+			LOG_INF("ISO buffer available (%u dropped)!", fail_cnt);
+		}
+
+		fail_cnt = 0;
```

Comment:
Ditto, different from IPC, but since then the logging subsystem has implemented rate limited versions like `LOG_INF_RATELIMIT` and `LOG_ERR_RATELIMIT` which was the main rationale behind the `fail_cnt` value